integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonAddLoadCommand.scala

```diff
@@ -92,6 +92,24 @@ case class CarbonAddLoadCommand(
     val segmentPath = options.getOrElse(
       "path", throw new UnsupportedOperationException("PATH is manadatory"))
 
+    val allSegments = SegmentStatusManager.readLoadMetadata(carbonTable.getMetadataPath)
+
+    // If a path is already added then we should block the adding of the same path again.
+    if (allSegments.exists(a =>
+      a.getPath != null && a.getPath.equalsIgnoreCase(segmentPath)
+    )) {
+      throw new AnalysisException(s"Cannot add the segment. This path is already in use by " +
```

Comment:
change message to `path already exists in table status file, can not add same segment path repeatedly: $path`

Reply:
Done